example/example.dart

```diff
@@ -8,23 +8,44 @@
 
 import 'package:dart_wot/dart_wot.dart';
 
+final Map<String, BasicCredentials> basicCredentials = {
+  'urn:test': BasicCredentials('rw', 'readwrite')
+};
+
+Future<BasicCredentials?> basicCredentialsCallback(
+  Uri uri,
+  Form? form, [
+  BasicCredentials? invalidCredentials,
+]) async {
+  final id = form?.interactionAffordance.thingDescription.identifier;
+
+  return basicCredentials[id];
```

Comment:
Due to `form?`, `id` can be null here. Will this lead to an exception, or does it return `null` (as the function signature indicates)?

Reply:
This should actually return `null`, see [this example](https://dartpad.dev/?id=e879483235a60237fdd7f87068dd63fe) as an illustration :)

Reply:
Ah, that's good to know, thanks for the example!